### PR TITLE
feat: add coin selection with BDK algorithms and manual UTXO control

### PR DESCRIPTION
**Pull request by coreyphillips**
## Summary

  This PR adds coin selection functionality to ldk-node, giving users control over which UTXOs are used in on-chain transactions. This is essential for fee optimization, proper UTXO management and privacy-conscious users.

  ## Key Features

  ### 1. UTXO Management
  - **List spendable outputs**: New `list_spendable_outputs()` method returns all UTXOs that are safe to spend
      - **Safety**: Automatically excludes UTXOs being used to fund Lightning channels

  ### 2. Coin Selection Algorithms
  Added `select_utxos_with_algorithm()` method with four algorithms:
  - **BranchAndBound**: Finds exact amount matches to minimize change
  - **LargestFirst**: Prioritizes largest UTXOs (reduces UTXO set size)
  - **OldestFirst**: FIFO selection (useful for accounting/tax purposes)
  - **SingleRandomDraw**: Random selection for privacy

  ### 3. Manual UTXO Control
  - Updated `send_to_address()` to accept optional `utxos_to_spend` parameter
  - Allows users to specify exactly which UTXOs to use in a given transaction

  ### Expected Usage
  - `list_spendable_outputs` -> `select_utxos_with_algorithm` -> `send_to_address`
  - `list_spendable_outputs` -> `<custom filter logic>` -> `send_to_address`

  ## API Changes

  ### New Methods
  ```rust
  // List all spendable UTXOs
  pub fn list_spendable_outputs() -> Result<Vec<SpendableUtxo>, Error>

  // Select UTXOs using an algorithm
  pub fn select_utxos_with_algorithm(
      target_amount_sats: u64,
      fee_rate: Option<FeeRate>,
      algorithm: CoinSelectionAlgorithm,
      utxos: Option<Vec<SpendableUtxo>>
  ) -> Result<Vec<SpendableUtxo>, Error>

  // Updated send_to_address with UTXO selection
  pub fn send_to_address(
      address: &Address,
      amount_sats: u64,
      fee_rate: Option<FeeRate>,
      utxos_to_spend: Option<Vec<SpendableUtxo>> // NEW parameter
  ) -> Result<Txid, Error>

  New Types

  - SpendableUtxo: Contains outpoint and value_sats
  - CoinSelectionAlgorithm: Enum with the four algorithms
  - Error::CoinSelectionFailed: New error for selection failures

  Use Cases

  1. Privacy: Users can avoid linking UTXOs from different sources
  2. Fee Optimization: Select UTXOs to minimize transaction size/fees
  3. Dust Management: Consolidate small UTXOs when fees are low
  4. Accounting: Use OldestFirst for FIFO tax calculations
  5. Channel Safety: Automatically protects channel funding UTXOs

  Testing Recommendations

  - Test each coin selection algorithm with various UTXO sets
  - Verify channel funding UTXOs are properly excluded
  - Test edge cases (insufficient funds, all UTXOs locked, etc.)
  - Verify manual UTXO selection works correctly

  Breaking Changes

  None - the changes are backwards compatible. The new parameter in send_to_address() is optional.

### CODE REVIEW

File: bindings/swift/Sources/LDKNode/LDKNode.swift [hunk 1]
```diff
@@ -5340,6 +5363,63 @@ public func FfiConverterTypeSendingParameters_lower(_ value: SendingParameters)
     return FfiConverterTypeSendingParameters.lower(value)
 }
 
+
+public struct SpendableUtxo {
+    public var outpoint: OutPoint
```
Comment (dzdidi) [inline bindings/swift/Sources/LDKNode/LDKNode.swift:5368]: What is `OutPoint`? Is it `TxId` + `vout`?

File: src/error.rs [hunk 2]
```diff
@@ -201,10 +203,15 @@ impl fmt::Display for Error {
...
+			Self::CannotRbfFundingTransaction => {
+				write!(f, "Cannot RBF a channel funding transaction.")
+			},
 			Self::TransactionNotFound => write!(f, "The transaction was not found in the wallet."),
-			Self::TransactionAlreadyConfirmed => write!(f, "The transaction is already confirmed and cannot be modified."),
-			Self::NoSpendableOutputs => write!(f, "The transaction has no spendable outputs.")
+			Self::TransactionAlreadyConfirmed => {
+				write!(f, "The transaction is already confirmed and cannot be modified.")
+			},
+			Self::NoSpendableOutputs => write!(f, "The transaction has no spendable outputs."),
```
Comment (dzdidi) [inline src/error.rs:213]: I know it was added a while back, but still. When can transaction have no spendable outputs?

Comment (coreyphillips) [reply src/error.rs:213]: When creating a transaction it's possible that some or all of the UTXO's passed in are used in lightning funding transactions. In which case the intended transaction does not have any spendable outputs.

**Review by ovitrif — COMMENTED** [review]
Can the updated bindings for `ldk-node-android` be committed as well?

Should work by running the existing script:
```sh
sh scripts/uniffi_bindgen_generate_kotlin_android.sh
```

If not please let me know so I can open a PR targeting the branch of this one 🙏🏻

**Comment by coreyphillips** [comment]
> Can the updated bindings for `ldk-node-android` be committed as well?
> 
> Should work by running the existing script:
> 
> ```shell
> sh scripts/uniffi_bindgen_generate_kotlin_android.sh
> ```
> 
> If not please let me know so I can open a PR targeting the branch of this one 🙏🏻

Updated [here](https://github.com/synonymdev/ldk-node/pull/6/commits/46e093df627f91f4eba07386f12508b284e3b1a6). Still adjusting to this flow so please let me know if I missed anything.
